### ISSUE TITLE
Don't include all rgws in ceph conf

### ISSUE THREAD

**Issue by clwluvw** [ceph/ceph-ansible#5579]
**Bug Report**

What happened:
In deploying rgw it will add all rgw hosts to ceph conf. It seems to be okay but when you run ansible with `--limit` it will add just the limitations and when you can't run ansible on all your nodes at a same time, each node will have different and messy ceph conf.

What you expected to happen:
Won't it better to no loop on rgws host here and just add each host config to it's ceph conf?
https://github.com/ceph/ceph-ansible/blob/master/roles/ceph-config/templates/ceph.conf.j2#L108
I mean for example host1 just have `client.rgw.host1` config instead of having both `host1` and `host2`?
This loop was added to solve #49 but I think it could be solve without loop on rgw instances!

How to reproduce it (minimal and precise):
deploy rgw with `--limit` on one or two hosts

**Comment by guits** [comment]
@clwluvw 
`{% for host in groups[rgw_group_name] %}` means we are iterating over all rgw nodes whatever we run with `--limit` or not so I'm not sure how you end up with a ceph.conf including only nodes included in `--limit`

**Comment by clwluvw** [comment]
@guits I'm using ansible 2.9.9 and it just pick rgws that included in `--limit`. Can you test it please?

**Comment by guits** [comment]
@clwluvw see below:

site.yml:
```yaml
- hosts: mons
  become: false

  tasks:
    - name: test
      debug:
        msg: "{% for host in groups['mons'] %} {{ hostvars[host]['ansible_hostname'] }} {% endfor %}"
```

hosts:
```ini
[mons]
mon0 ansible_host=mon0
mon1 ansible_host=mon1
mon2 ansible_host=mon2
mon3 ansible_host=mon3
mon4 ansible_host=mon4
```

running with `ansible-playbook -i hosts site.yml --limit mon0,mon1` :
```
TASK [test] *************************************************************************************************************************************************************************
Friday 24 July 2020  08:22:58 +0200 (0:00:00.130)       0:00:05.958 ***********
ok: [mon0] =>
  msg: ' mon0  mon1  mon2  mon3  mon4 '
ok: [mon1] =>
  msg: ' mon0  mon1  mon2  mon3  mon4 '

PLAY RECAP **************************************************************************************************************************************************************************
mon0                       : ok=3    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
mon1                       : ok=3    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```


```
$ ansible-playbook --version
ansible-playbook 2.9.9
```

**Comment by clwluvw** [comment]
@guits I know and I see this behavior in `set_fact monitor_name ansible_hostname` task but in that loop it just pick `--limit` rgws. Maybe I should run this before deploying? Did you try with deploying rgws and get the same?
```yaml
- name: gather and delegate facts
      setup:
      delegate_to: "{{ item }}"
      delegate_facts: True
      with_items: "{{ groups['all'] | difference(groups.get('clients', [])) }}"
      run_once: true
      when: delegate_facts_host | bool
```

**Comment by clwluvw** [comment]
@guits Also thanks for this PR #5587. This is more reasonable than including all rgws in ceph conf.

**Comment by guits** [comment]
@clwluvw I've tried with same conditions and I'm still unable to reproduce.

anyway, I've sent #5587 which should address your issue...

**Comment by clwluvw** [comment]
@guits Yes it's okay. I'm really happy with your PR. I'll be wait for `stable-4.0` backport. Thanks.

**Comment by clwluvw** [comment]
@guits Do you have any due date for backport it to `stable-4.0`?

**Comment by guits** [comment]
@clwluvw I need to figure out a couple of details before merging it to master first, then the backport to stable-4.0 should follow pretty quickly.
I can't give you an accurate due date for now but it shouldn't be that long.
Thanks for your patience!

**Comment by guits** [comment]
closed by #5587
